mitodl/ocw-studio
# Added front-end for configuring navigation menus

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #340

#### What's this PR do?
- Adds the front-end portion for configuring navigation menus 
- Adds the [`react-nestable`](https://github.com/primetwig/react-nestable) package to support drag and drop

#### How should this be manually tested?
- Update your local configs (`manage.py override_site_config`)
- Create a new site or select an existing site with the OCW course site starter
- Select Settings > Menu, and click "Add New" to add pages/resources to the navigation menu

From here, you'll want to test several things:
- You can add pages/resources _or_ external links as menu items
- You can reorder and nest these items freely
- You can update individual items using the gear icon
- You can remove items using the trash bin icon
- Changes are only permanently applied if you click "Save"
- The exact same menu is shown if you refresh the page (if the menu was saved)
- The data that is saved for the menu is in a format that is compatible with the [Hugo menu format](https://gohugo.io/content-management/menus/#add-non-content-entries-to-a-menu), minus the `url` property for internal content links as those will be filled in when site content is pushed to Github 

#### Where should the reviewer start?
`static/js/components/widgets/MenuField.tsx`

#### Any background context you want to provide?
- To speed things up, this is being done in a Hugo-specific way. By that I mean the data structure that is being used to save the menus is in the format that Hugo expects ([doc](https://gohugo.io/content-management/menus/#add-non-content-entries-to-a-menu))

#### Screenshots (if appropriate)
![ss 2021-06-23 at 17 53 49 ](https://user-images.githubusercontent.com/14932219/123328534-b0a5a600-d509-11eb-80ca-2aef66020f2e.png)
![ss 2021-06-24 at 16 31 32 ](https://user-images.githubusercontent.com/14932219/123328542-b4392d00-d509-11eb-8749-7738b64c901a.png)
![ss 2021-06-24 at 16 32 37 ](https://user-images.githubusercontent.com/14932219/123328643-d5018280-d509-11eb-9582-11125fcf7e13.png)
![ss 2021-06-23 at 18 05 42 ](https://user-images.githubusercontent.com/14932219/123328556-b69b8700-d509-11eb-95f2-94c52e4d87eb.png)
